Lucene.Net.DocumentMapper/FieldMappers/StringFieldMapper.cs

```diff
@@ -15,17 +15,29 @@ public bool IsMatch(PropertyInfo propertyInfo)
             return type == typeof(string) || type == typeof(object) || type == typeof(String) || type == typeof(Object);
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="propertyInfo"></param>
+        /// <param name="value">System.String value</param>
+        /// <param name="name">field name</param>
+        /// <returns></returns>
         public Field MapToField(PropertyInfo propertyInfo, object value, string name)
         {
-            if (value.ToString().Length > 32766)
-            {
-                return new Field(name, value.ToString(), GetStore(propertyInfo), Field.Index.NO);
-            }
+            //if (value.ToString().Length > 32766)
```

Comment:
I can't remember why I did this, I think it was because once the string gets to a certain length it can't tokenize it, is that updated in newer versions of Lucene?

Reply:
if you use lucene.net 4.8.0-beta00016
```
        [Obsolete("Use StringField, TextField instead.")]
        public Field(string name, string value, Store store, Index index)
            : this(name, value, TranslateFieldType(store, index, TermVector.NO))
        {
        }
```
ObsoleteAttribute marks **Field** that are no longer in use. 

Creates a new Lucene.Net.Documents.StringField (a field that is indexed but not tokenized)

```
    //     A field that is indexed but not tokenized: the entire System.String value is
    //     indexed as a single token. For example this might be used for a 'country' field
    //     or an 'id' field, or any field that you intend to use for sorting or access through
    //     the field cache.
public StringField(string name, string value, Store stored);
```

Creates a new Lucene.Net.Documents.TextField with System.String value.(a field that is indexed and tokenized)

```
    //     A field that is indexed and tokenized, without term vectors. For example this
    //     would be used on a 'body' field, that contains the bulk of a document's text.

public TextField(string name, string value, Store store)
            : base(name, value, (store == Store.YES) ? TYPE_STORED : TYPE_NOT_STORED)
```

The field is indexed but not tokenized -> StringField
The field is indexed and tokenized -> TextField

`return new Field(name, value.ToString(), GetStore(propertyInfo), Field.Index.NO);`

Reply:
Okay I can try to update this, but I'm pretty sure if it gets to that length you can't index it either